src/com/palmergames/bukkit/towny/db/TownySQLSource.java

```diff
@@ -77,41 +80,46 @@ public TownySQLSource(Towny plugin, TownyUniverse universe, String type) {
 		/*
 		 * Setup SQL connection
 		 */
-		String hostname = TownySettings.getSQLHostName();
-		String port = TownySettings.getSQLPort();
-		String flags = TownySettings.getSQLFlags();
 		db_name = TownySettings.getSQLDBName();
 		tb_prefix = TownySettings.getSQLTablePrefix().toUpperCase();
 		
-		String driver1;
-		if (this.type.equals("h2")) {
-
-			driver1 = "org.h2.Driver";
-			this.dsn = ("jdbc:h2:" + dataFolderPath + File.separator + db_name + ".h2db;AUTO_RECONNECT=TRUE");
-			username = "sa";
-			password = "sa";
-
-		} else if (this.type.equals("mysql")) {
-
-			driver1 = "com.mysql.jdbc.Driver";
-			this.dsn = ("jdbc:mysql://" + hostname + ":" + port + "/" + db_name + flags);
-			username = TownySettings.getSQLUsername();
-			password = TownySettings.getSQLPassword();
-
-		} else {
-
-			driver1 = "org.sqlite.JDBC";
-			this.dsn = ("jdbc:sqlite:" + dataFolderPath + File.separator + db_name + ".sqldb");
-			username = "";
-			password = "";
-
-		}
+		this.dsn = ("jdbc:mysql://" + TownySettings.getSQLHostName() + ":" + TownySettings.getSQLPort() + "/" + db_name + TownySettings.getSQLFlags());
+		this.config = new HikariConfig();
+		
+		config.setPoolName("Towny MySQL");
+		config.setJdbcUrl(this.dsn);
+
+		username = TownySettings.getSQLUsername();
+		password = TownySettings.getSQLPassword();
+
+		config.setUsername(username);
+		config.setPassword(password);
+
+		config.addDataSourceProperty("cachePrepStmts", "true");
+		config.addDataSourceProperty("prepStmtCacheSize", "250");
+		config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+		config.addDataSourceProperty("useServerPrepStmts", "true");
+		config.addDataSourceProperty("useLocalSessionState", "true");
+		config.addDataSourceProperty("rewriteBatchedStatements", "true");
+		config.addDataSourceProperty("cacheResultSetMetadata", "true");
+		config.addDataSourceProperty("cacheServerConfiguration", "true");
+		config.addDataSourceProperty("elideSetAutoCommits", "true");
+		config.addDataSourceProperty("maintainTimeStats", "false");
+		config.addDataSourceProperty("alwaysSendSetIsolation", "false");
+		config.addDataSourceProperty("cacheCallableStmts", "true");
+
+		config.setMaximumPoolSize(TownySettings.getMaxPoolSize());
+		config.setMinimumIdle(TownySettings.getMinIdleTime());
```

Comment:
`setMinimumIdle` represents the number of minimum connections that Hikari will maintain, and **not** a timeout for each connections. So I think the config setting should be renamed to `MinIdleConnections`. Although, hikari recommends not to set this setting. 

From HikariCP README: 
> This property controls the minimum number of idle connections that HikariCP tries to maintain in the pool. If the idle connections dip below this value and total connections in the pool are less than maximumPoolSize, HikariCP will make a best effort to add additional connections quickly and efficiently.

Reply:
Solved this by just removing it since it probably is not needed.